### ISSUE TITLE
Pipelines.show method typescript signature is not complete

### ISSUE THREAD

**Issue by lionel87** [jdalrymple/gitbeaker#3122]
**Description**

The Gitlab Pipelines API allows the user to get the latest pipeline info with placing the `latest` string in place of a pipeline id.
With Gitbeaker I can achieve this with:

```js
gitlab.Pipelines.show(project, 'latest');
```

My problem is when I use typescript, putting the `latest` string there throws type error, it expects `number` there.

Also here, while using the `Pipelines.show(project, pipelineid, options)` format, the options type is missing the ref property which allows me to specify a branch name:
```js
gitlab.Pipelines.show(project, 'latest', { ref: 'develop' });
```
Errors out with: `Object literal may only specify known properties, and 'ref' does not exist in type 'Sudo'.`

**Possible solutions**

It would be a solution to allow a `'latest' | number` type at the pipelineId property, and add `{ ref: string; }` property to options.

My current workaround, which works, but ugly:
```ts
gitlab.Pipelines.show(project, 'latest' as any, { ref: 'develop' } as any);
```

Possibly no code modifications needed, only type definitions affected since this example works correctly.

**Checklist**
- [x] I have checked that this is not a duplicate issue.
- [x] I have read the documentation.

**Comment by jdalrymple** [comment]
Give the next release a try, it changed the typing of the pipelineId to be string | number. As for the options, where in the docs does it show you can pass ref when getting a single pipeline?

**Comment by lionel87** [comment]
Thanks for the info I will update today and try it out.

I seen the `ref` parameter here: [Get the latest pipeline](https://docs.gitlab.com/ee/api/pipelines.html#get-the-latest-pipeline)

I think the ref is only meaningful if you provide 'latest' to the pipeline id, eg. get the latest pipeline for a specific branch or tag.

**Comment by jdalrymple** [comment]
I am apparently blind haha, ill fix that up!

**Comment by jdalrymple** [comment]
Give the latest canary build a look, it should fix this issue!

**Comment by jdalrymple** [comment]
:rocket: Issue was released in [`37.0.0`](https://github.com/jdalrymple/gitbeaker/releases/tag/37.0.0) :rocket:
